### ISSUE TITLE
Yolov5 classifier with training mnist dataset results were weird

### ISSUE THREAD

**Issue by chiehpower** [ultralytics/yolov5#5077]
## ❔Question
hi all, 
I was trying to use yolov5 to do a classification that I just trained the default dataset, mnist, to check its performance. 
However, I got these so weird accuracy results. 

![2021-10-0755](https://user-images.githubusercontent.com/32332200/136362435-3cb57e9f-6f62-4961-8315-118c4dd8abff.png)

BTW I did not change any hyperparameters, only adjusted the batch size.

Is there any idea about this problem?  

Thank you so much!

**Comment by glenn-jocher** [comment]
@chiehpower you have loss instabilities leading to nan loss, which is unrecoverable. Once this happens there is no point in continuing training. You should perhaps return to the original settings that allowed the dataset to train stably.

**Comment by chiehpower** [comment]
> @chiehpower you have loss instabilities leading to nan loss, which is unrecoverable. Once this happens there is no point in continuing training. You should perhaps return to the original settings that allowed the dataset to train stably.

Hi @glenn-jocher ,

Thanks for your reply!
I have resolved the problem!! 
Now it can work well! 

BR,
Chieh

**Comment by mengmeng0406** [comment]
> > @chiehpower you have loss instabilities leading to nan loss, which is unrecoverable. Once this happens there is no point in continuing training. You should perhaps return to the original settings that allowed the dataset to train stably.
> 
> Hi @glenn-jocher ,
> 
> Thanks for your reply! I have resolved the problem!! Now it can work well!
> 
> BR, Chieh
I got the same results. Could you tell me how you solved this problem?

**Comment by glenn-jocher** [comment]
@mengmeng0406 I've updated the classifier branch recently and added `--adam` to the Usage example as it showed better stability. You should get stable training with the latest code just using the default Usage example, 99% accuracy in 4 epochs:

```
python path/to/classifier.py --model yolov5s --data mnist --epochs 5 --img 128 --adam
```

<img width="1136" alt="Screen Shot 2021-12-09 at 1 05 30 PM" src="https://user-images.githubusercontent.com/26833433/145393455-48c46f8c-7e65-49b3-86b6-95d1ce59b3c2.png">
